guides/v2.2/ext-best-practices/tutorials/dynamic-row-system-config.md

```diff
@@ -56,14 +64,113 @@ class Ranges extends AbstractFieldArray
         $this->addColumn('from_qty', ['label' => __('From'), 'class' => 'required-entry']);
         $this->addColumn('to_qty', ['label' => __('To'), 'class' => 'required-entry']);
         $this->addColumn('price', ['label' => __('Price'), 'class' => 'required-entry']);
+        $this->addColumn('tax', [
+            'label' => __('Tax'),
+            'renderer' => $this->getTaxRenderer()
+        ]);
         $this->_addAfter = false;
         $this->_addButtonLabel = __('Add');
     }
+
+    /**
+     * Prepare existing row data object
+     *
+     * @param DataObject $row
+     * @throws LocalizedException
+     */
+    protected function _prepareArrayRow(DataObject $row): void
+    {
+        $options = [];
+
+        $tax = $row->getTax();
+        if ($tax !== null) {
+            $options['option_' . $this->getTaxRenderer()->calcOptionHash($tax)] = 'selected="selected"';
+        }
+
+        $row->setData('option_extra_attrs', $options);
+    }
+
+    /**
+     * @return TaxColumn
+     * @throws LocalizedException
+     */
+    private function getTaxRenderer()
+    {
+        if (!$this->taxRenderer) {
+            $this->taxRenderer = $this->getLayout()->createBlock(
+                TaxColumn::class,
+                '',
+                ['data' => ['is_render_to_js_template' => true]]
+            );
+        }
+        return $this->taxRenderer;
+    }
 }
 ```
 
 This block prepares the desired columns for inclusion in the new config.
 
+## Step 3: Create the block class to describe a column with drop-down input
+
+> `File: app/code/Vendor/Module/Block/Adminhtml/Form/Field/TaxColumn.php`
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace Vendor\Module\Block\Adminhtml\Form\Field;
+
+use Magento\Framework\View\Element\Html\Select;
+
+class TaxColumn extends Select
+{
+    /**
+     * Set "name" for <select> element
+     *
+     * @param string $value
+     * @return $this
+     */
+    public function setInputName($value)
+    {
+        return $this->setName($value);
+    }
+
+    /**
+     * Set "id" for <select> element
+     *
+     * @param $value
+     * @return $this
+     */
+    public function setInputId($value)
+    {
+        return $this->setId($value);
+    }
+
+    /**
+     * Render block HTML
+     *
+     * @return string
+     */
+    public function _toHtml(): string
```

Comment:
It's not a big deal, but should we override the parent method scope of `_toHtml()`? It's `protected` in the parent class.

Reply:
@rogyar , I see such approach in the existing Magento classes: 
- [Magento\CatalogInventory\Block\Adminhtml\Form\Field\Customergroup::_toHtml()](https://github.com/magento/magento2/blob/2.3.3/app/code/Magento/CatalogInventory/Block/Adminhtml/Form/Field/Customergroup.php#L106)
- [Magento\Braintree\Block\Adminhtml\Form\Field\Countries::_toHtml()](https://github.com/magento/magento2/blob/2.3.3/app/code/Magento/Braintree/Block/Adminhtml/Form/Field/Countries.php#L40)
- [Magento\Braintree\Block\Adminhtml\Form\Field\CcTypes::_toHtml()](https://github.com/magento/magento2/blob/2.3.3/app/code/Magento/Braintree/Block/Adminhtml/Form/Field/CcTypes.php#L43)

So, in my PR I've followed the same logic.

Thanks.

Reply:
I believe it's a sort of mechanical mistake. But, not a big deal, as I said.